### PR TITLE
points and vectors based on arithmetic operations

### LINKED ISSUE #1872: Request: points and vectors based on arithmetic operations
We should be able to create points and vector from arithmetic operations involving points, vectors, and maths.

Some examples that should work.
```
<vector name="v">(1,2)</vector>
<point name="P">(3,4)</point>
<math name="m">(7,5)</math>
<number name="n">6</number>
<vector>2$v</vector>
<point>$v + $n$P + $m - (3,9)</point>
<polygon name="triangle" vertices="$v+$m $n(4,2) 5$P" />
```

### PR DESCRIPTION
**Pull request by dqnykamp**
Resolves #1872

### CODE REVIEW

**Comment by dqnykamp** [comment]
Note: this is a breaking change in that point lists must now be space separated.  Previously, `<line through="(1,2)(3,4)" />` would create a line through the points `(1,2)` and `(3,4)`.  With the changes in this PR, that DoenetML no longer specifies a valid line as, without the space, `(1,2)(3,4)` is treated a single (invalid) point.

**Comment by dqnykamp** [comment]
Need to address points in a point list coming from composites. They are handling incorrectly.

**Comment by dqnykamp** [comment]
points in a point list coming from composites are now addressed

File: src/Core/components/MathList.js [hunk 1]
```diff
@@ -55,29 +52,13 @@ export default class MathList extends InlineComponent {
     let sugarInstructions = super.returnSugarInstructions();
 
     let groupIntoMathsSeparatedBySpaces =
-      returnGroupIntoComponentTypeSeparatedBySpaces({
-        componentType: "math",
-      });
-    let breakStringsIntoMathsBySpaces =
-      returnBreakStringsIntoComponentTypeBySpaces({
+      returnGroupIntoComponentTypeSeparatedBySpacesOutsideParens({
```
Comment (jaltekruse) [inline src/Core/components/MathList.js:55]: holy descriptive name batman

**Review by jaltekruse — APPROVED** [review]
+1

File: src/Core/components/MathList.js (hunk 1 above)
Comment (dqnykamp) [reply src/Core/components/MathList.js:55]: Hah!  Yeah, I kept getting confused by different break strings functions, so this is my effort to keep them apart.

**Pull request merged**